models/DataObject/ClassDefinition/Data/Block.php

```diff
@@ -554,7 +554,10 @@ public function hasChildren(): bool
         return false;
     }
 
-    public function addChild(Data|Layout $child)
+    /**
+     * @param Data|Layout $child
+     */
+    public function addChild(mixed $child)
```

Comment:
@dvesh3 should we add a comment here (and all the other places) why `mixed` is necessary?

Reply:
yes, this is required for assets metadata definitions bundle. please see https://github.com/pimcore/asset-metadata-class-definitions/actions/runs/3576806963/jobs/6015056604#step:10:47

I'll add comments here once the approach is finalized.

Reply:
done here https://github.com/pimcore/pimcore/pull/13705/commits/cbbbe8150108d3b2a1387eae678e1bd103998dc4